dev-docs/modules/userId.md

```diff
@@ -2349,6 +2348,72 @@ If you're an ID provider that wants to get on this page:
+
+Please find more details [Share encrypted signals with bidders (Beta)](https://support.google.com/admanager/answer/10488752?hl=en)
+
+{: .table .table-bordered .table-striped }
+| Param under userSync | Scope | Type | Description | Example |
+| --- | --- | --- | --- | --- |
+| encryptedSignal | Optional | Object | Publisher can specify the ESP config by adding encryptedSignal Object under userSync Object |  |
+| encryptedSignal.eids | Required | Object |  Object consist of sources list and encryption flag | Check below config as an example  |
+| encryptedSignal.eids.sources | Required | Array | An array of sources for which signals needs to be registered  | `['sharedid.org','criteo.com']` |
+| encryptedSignal.eids.encrypt | Required | Boolean | Should be set to false by default. Please find below note | `true` or `false` |
+| encryptedSignal.custom | Optional | Object | Object consist of Source list and its associated function to retrieve the data  |  |
+| encryptedSignal.custom.sources | Required | Array of Object | Object consist of source list and respective custom function  | Check below config as an example |
+| encryptedSignal.custom.sources.source | Required | Array  | List of custom sources for which signals needs to be registered   | Check below config as an example |
+| encryptedSignal.custom.sources.customFunc | Required | function | Function will be called which will return the custom data set from the page  | Check below config as an example  |
+| encryptedSignal.custom.encrypt | Required | Boolean | Default to true, Custom data will be encoded by Base64. (Currently base64 encode is enabled by default) | `true` | 
+| encryptedSignal.registerDelay | Optional | Integer | The amount of time (in seconds) after which registering of signals will happen. Default value 0 is considered if 'registerDelay' is not provided. |  `3000`
+
+{: .alert.alert-info :}
+**NOTE:**
+For eids encryption (encryptedSignal.eids.encrypt) set to true is not recommended unless downstream is informed of the changes.
+
+ESP Configuration Example:
+
+```
+pbjs.setConfig({
+    userSync: {
+        ...,
+        encryptedSignal: {
+            // Eids sources for which signals needs to registered.
+            "eids": {
+                "sources": [
+                    "sharedid.org",
+                    "criteo.com",
+                    "id5-sync.com",
+                    "pubcid.org",
+                    "audigent.com"
+                ],
+                "encrypt": false
+            },
+            // Custom Sources having custom function which returns respective data
+            "custom": {
+                "sources": [{
+                    source: ['pubmatic.com'],
+                    customFunc: () => {
+                        return '{"keywords":["tech","auto"]}';
```

Comment:
how do we pass, for example, pubmatic's id to pubmatic? if we fail to do so, does pubmatic end up with less of an ob footprint in cookied environments? In other words, should i restrict this to firefox/safari if I can't get the pubmatic id on this as well? 

Please add discussion of this q to the general doc.

Reply:
@patmmccann  If you are referring to SSP 3p cookies (till they exist) then that should be handled separately where SSP should work with OB/GAM to handles SSP specific id & signals. I don’t think pubs will in general have access to SSP 3p data on page.

I would recommend publishers reaching out to OB integrated SSP to see how that can be enabled for the integration.

@nitin0610 Can we add a note that : Publishers enabling passing eids/signal through ESP should reach out to SSPs integrated through OB to make sure to take any additional steps needed to ensure impact on 3p cookie based transaction is handled and impact is minimal.